### PR TITLE
dev-python/astropy: Version bump to 1.0.4

### PR DESCRIPTION
**Pull request by sargas**
This is to satisfy https://bugs.gentoo.org/556884 , my request to bump astropy to 1.0+.

I'm not sure I can ping any teams; @Gentoo/sci-astronomy is where it should go.
@jlec and @jauhien have merged my PRs into the science overlay, while @xarthisius and @sfabbro appear to have worked on astronomy bugs in Gentoo. Sorry for the spam to anyone who no longer works on this!

### CODE REVIEW

**Comment by jlec** [comment]
let me take a look into this.

File: dev-python/astropy/astropy-1.0.4.ebuild [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+IUSE="doc test"
+
+RDEPEND="
+	dev-libs/expat:0=
+	dev-python/numpy[${PYTHON_USEDEP}]
+	sci-astronomy/erfa:0=
```
Comment (jlec) [inline dev-python/astropy/astropy-1.0.4.ebuild]: needs at least 1.2.0

File: dev-python/astropy/astropy-1.0.4.ebuild [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	>=sci-astronomy/wcslib-4.23:0=
```
Comment (jlec) [inline dev-python/astropy/astropy-1.0.4.ebuild]: needs at least 4.25.1

**Comment by jlec** [comment]
Please try to unbundle wcslib, erfa and the  helpers package. Try to delete them in src_prepare and see what happens then. In my test the build system tries to use header files from the bundled versions instead of the system libs. Perhaps the --use-system-libs flag is not working correctly.

**Comment by sargas** [comment]
How do you know it's using the bundled libs? On here, both of:
```
   # ebuild astropy-1.0.4.ebuild clean prepare; rm -rf /var/tmp/portage/dev-python/astropy-1.0.4/work/astropy-1.0.4/cextern/; ebuild astropy-1.0.4.ebuild compile
```
or:
```
   # ebuild astropy-1.0.4.ebuild clean prepare; ebuild astropy-1.0.4.ebuild compile
```
give me warnings from /usr/include/erfa.h, used by the wrapper in `astropy/_erfa`. I don't see any sign that the bundled libs are being used.


You are right about astropy_helpers being required, however. I added a commit for debian's patch which disables the included astropy_helpers from being used, as well as another updating the deps to the bundled version. Thanks for updating those in the tree today.

**Comment by jlec** [comment]
The following code is a way to make sure the bundled libs aren't used.
```
python_prepare_all() {
    rm -r astropy_helpers || die
    rm -r cextern || die
    distutils-r1_python_prepare_all
}
```

File: dev-python/astropy/astropy-1.0.4.ebuild [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		dev-libs/libxml2[${PYTHON_USEDEP}]
+		dev-python/h5py[${PYTHON_USEDEP}]
+		dev-python/matplotlib[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+		sci-libs/scipy[${PYTHON_USEDEP}]
+	)"
+
+python_prepare_all() {
+	distutils-r1_python_prepare_all
+	epatch "${FILESDIR}/${P}-disable_helper.patch"
```
Comment (jlec) [inline dev-python/astropy/astropy-1.0.4.ebuild]: Just define PATCHES=() in global scope and drop the whole function.

**Comment by jlec** [comment]
I tried these changes

```
diff --git a/dev-python/astropy/astropy-1.0.4.ebuild b/dev-python/astropy/astropy-1.0.4.ebuild
index ee17b68..9bf6971 100644
--- a/dev-python/astropy/astropy-1.0.4.ebuild
+++ b/dev-python/astropy/astropy-1.0.4.ebuild
@@ -42,9 +42,14 @@ DEPEND="${RDEPEND}
        sci-libs/scipy[${PYTHON_USEDEP}]
    )"

+PATCHES=( "${FILESDIR}/${P}-disable_helper.patch" )
+
 python_prepare_all() {
+   rm -r ${PN}_helpers || die
+   rm -r cextern/*/* || die
+   rm ah_bootstrap.py || die
+   sed -e '/import ah_bootstrap/d' -i setup.py || die "Removing ah_bootstrap failed"
    distutils-r1_python_prepare_all
-   epatch "${FILESDIR}/${P}-disable_helper.patch"
 }

 python_compile() {
```

And got this error

```
Traceback (most recent call last):
  File "setup.py", line 57, in <module>
    package_info = get_package_info()
  File "/usr/lib64/python3.3/site-packages/astropy_helpers/setup_helpers.py", line 541, in get_package_info
    package_data.update(setuppkg.get_package_data())
  File "astropy/wcs/setup_package.py", line 322, in get_package_data
    shutil.copy(source, dest)
  File "/usr/lib64/python3.3/shutil.py", line 227, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib64/python3.3/shutil.py", line 109, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'cextern/wcslib/C/cel.h'
```

That sounds like the buildsystem tries to copy headers from the bundled libs.

**Comment by sargas** [comment]
During the install you're right, it seems `python_install` needs the `use_system_libraries` too. 

During compiling, I can't explain why, but it seems to happen when you remove ah_bootstrap and the import statement in `python_prepare_all`. It looks like distutils loses all its command_options without ah_bootstrap. I don't see anything in ah_bootstrap.py that accounts for this though.

**Comment by jlec** [comment]
@sargas please check the installed astropy_helpers package. There should be something.

**Comment by monsieurp** [comment]
What's the status of this PR?

**Pull request closed without merging**